src/azul/queues.py

```diff
@@ -209,60 +207,19 @@ def _get_queue_lengths(self, queues: Mapping[str, Queue]) -> Tuple[int, Mapping[
-                             min_timeout: Optional[int] = None,
-                             max_timeout: Optional[int] = None):
+    def wait_to_stabilize(self) -> int:
         """
-        Wait for the work queues to reach a desired fill level.
-
-        :param empty: If True, wait for the queues to drain (be empty).
-                      If False, wait until they are not emtpy.
-
-        :param num_expected_bundles: Number of bundles being indexed. If None,
-                                     the number of bundles will be approximated
-                                     dynamically based on the number of message
-                                     in the notifications queue.
-
-        :param min_timeout: Minimum timeout in seconds. If specified, wait at
-                            least this long for queues to reach the desired
-                            level.
-
-        :param max_timeout: Maximum timeout in seconds. If specified, wait at
-                            most this long for queues to reach the desired
-                            level.
+        Wait for the work queues to reach a steady state.
         """
-        require(min_timeout is None or max_timeout is None or min_timeout <= max_timeout,
-                'min_timeout must be less than or equal to max_timeout',
-                min_timeout, max_timeout)
-
-        def limit_timeout(timeout):
-            if max_timeout is not None and max_timeout < timeout:
-                timeout = max_timeout
-            if min_timeout is not None and timeout < min_timeout:
-                timeout = min_timeout
-            return timeout
-
-        timeout = limit_timeout(2 * 60)
         sleep_time = 5
         queues = self.get_queues(config.work_queue_names)
-        if empty:
-            total_lengths = deque(maxlen=12)
-            # One minute allotted for eventual consistency on SQS.
-            # For more info, read WARNING section on
-            # https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sqs.html#SQS.Client.get_queue_attributes
-            assert 12 * sleep_time >= 60
-        else:
-            total_lengths = deque(maxlen=1)
-
-        start_time = time.time()
-        num_bundles = 0
+        total_lengths = deque(maxlen=15)
+        # One minute allotted for eventual consistency on SQS.
+        # For more info, read WARNING section on
+        # https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sqs.html#SQS.Client.get_queue_attributes
+        assert total_lengths.maxlen * sleep_time >= 60
 
-        logger.info('Waiting for %s queues to %s notifications about %s bundles ...',
-                    len(queues),
-                    'be drained of' if empty else 'fill with',
-                    'an unknown number of' if num_expected_bundles is None else num_expected_bundles)
+        logger.info('Waiting for %s queue(s) to stabilize ...', len(queues))
```

Comment:
Continued from https://github.com/DataBiosphere/azul/pull/2855#discussion_r602057505

> That wouldn't be equivalent since the loop would sleep _before_ subsequent log statements instead of after. Instead of `log, [check, log, sleep], [check, log, sleep] ...` it would be `[log, check, sleep], [log, check, sleep]...`

It doesn't need to be 100% equivalent if the current implementation does not make sense. And the `log check log …` at the beginning of the sequence in the current implementation does not. Consider making the sequence  `check (log sleep check)*`.